From cb37ba19caf8dfa593002b14ed5d7b0ac3c5bfe4 Mon Sep 17 00:00:00 2001
From: Tal Yacobi <tyacobi@habana.ai>
Date: Wed, 14 May 2025 14:41:25 +0300
Subject: [PATCH] [SW-228793] pldm: translate pstate to pldm state set

Current translation returned the internal pstate
enum value which is different from the pldm power
performance state set value.

Translate the internal pstate value to the
pldm power performance state set value.

Change-Id: I65edf53ed65d4e070ab0928d9efc7caec7b77657
Signed-off-by: Tal Yacobi <tyacobi@habana.ai>
---

diff --git a/arch/gaudi3/ras/pldm/sensord_handlers.c b/arch/gaudi3/ras/pldm/sensord_handlers.c
index 66e4fef..e0c58b5 100644
--- a/arch/gaudi3/ras/pldm/sensord_handlers.c
+++ b/arch/gaudi3/ras/pldm/sensord_handlers.c
@@ -1954,7 +1954,7 @@
 	struct hl_nvme_mi_hbm_throt *hbm_throttling = hl_get_hbm_throttling_info();
 
 	sample->composite[COMP_SENSOR_DEVICE_POWER_PERFORMANCE_STATE] =
-		(p_state == PSTATE_UNKNOWN) ? PSTATE_HIGHEST : p_state;
+		(p_state == PSTATE_UNKNOWN) ? 0 : p_state + 1;
 
 	sample->composite[COMP_SENSOR_EMERGENCY_BREAK_STATE] =
 		hl_get_power_break_state() ?
